libminifi/include/utils/ByteArrayCallback.h

```diff
@@ -65,7 +66,7 @@ class ByteInputCallBack : public InputStreamCallback {
   }
 
   virtual char *getBuffer(size_t pos) {
-    return ptr + pos;
+    return reinterpret_cast<char*>(&vec[0]) + pos;
```

Comment:
Please use `vec.data()` to avoid out-of-bounds indexing when `vec` is empty, or index with pos.
```suggestion
    return &vec[pos];
```
or
```suggestion
    return vec.data() + pos;
```

Reply:
Done in [839ed47](https://github.com/apache/nifi-minifi-cpp/pull/897/commits/839ed47d2c3618a1fb1443f0929bfc8908051a87)